### PR TITLE
Graduate Configuration API version to v1beta1

### LINKED ISSUE #639: Graduate ComponentConfig API version to beta
**What would you like to be added**:
I would like to graduate the `config.kueue.x-k8s.io` to `v1beta1`.

**Why is this needed**:
Currently, we use v1beta1 API for `kueue.x-k8s.io` (e.g., clusterQueue), and the API is production ready.
So it is better to graduate also `config.kueue.x-k8s.io` to beta API so that we declare the whole of Kueue is production ready. 

Maybe, there is no need to make any breaking changes to the `config.kueue.x-k8s.io`, unlike when we migrated `kueue.x-k8s.io` from v1alpha2 to v1beta1.

**Completion requirements**:

This enhancement requires the following artifacts:

- [ ] Design doc
- [x] API change
- [x] Docs update

The artifacts should be linked in subsequent comments.

### PR DESCRIPTION
**Pull request by tenzen-y**
#### What type of PR is this?

/kind feature
/kind api-change

#### What this PR does / why we need it:
I changed the Configuration API version to v1beta1.

#### Which issue(s) this PR fixes:

Fixes #639

### CODE REVIEW

File: apis/config/v1beta1/configuration_types.go [hunk 1]
```diff
@@ -1,5 +1,5 @@
 /*
-Copyright 2022 The Kubernetes Authors.
+Copyright 2023 The Kubernetes Authors.
```
Comment (alculquicondor) [inline apis/config/v1beta1/configuration_types.go:2]: revert

The copyright year should be the year when the file was first added.

Comment (tenzen-y) [reply apis/config/v1beta1/configuration_types.go]: Makes sense. Reverted.

File: apis/config/v1beta1/groupversion_info.go [hunk 2]
```diff
@@ -14,10 +14,10 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-// Package v1alpha2 contains API Schema definitions for the config v1alpha2 API group
+// Package v1beta1 contains API Schema definitions for the config v1beta1 API group
 // +kubebuilder:object:generate=true
 // +groupName=config.x-k8s.io
```
Comment (tenzen-y) [inline apis/config/v1beta1/groupversion_info.go:19]: @alculquicondor Which group name is corrected? `config.x-k8s.io` vs `config.kueue.x-k8s.io`?

Comment (alculquicondor) [reply apis/config/v1beta1/groupversion_info.go:19]: `config.kueue.x-k8s.io`

Comment (tenzen-y) [reply apis/config/v1beta1/groupversion_info.go]: Thanks. I updated this line.

**Comment by alculquicondor** [comment]
Separate PR for docs?

/lgtm
/approve

/hold

**Comment by tenzen-y** [comment]
> Separate PR for docs?

Yes, I'd like to work for docs in a separate PR.

**Comment by tenzen-y** [comment]
@alculquicondor Can I unhold this PR?
I will update the docs in a separate PR.

**Comment by alculquicondor** [comment]
oh yeah, I put the hold in case you wanted to fix in this PR

/hold cancel

**Pull request merged**